django/db/models/indexes.py

```diff
@@ -0,0 +1,89 @@
+import hashlib
+
+from django.db.backends.utils import truncate_name
+from django.utils.encoding import force_bytes
+
+
+__all__ = ['Index']
+
+
+class Index(object):
+    def __init__(self, fields=None, field_names=None, model=None):
+        self.fields = fields
+        self.field_names = tuple(field_names) if field_names else ()
+        self.model = model
+        if self.fields:
+            self.field_names = tuple(f.name for f in fields)
+
+    def _normalize_fields(self):
+        if self.fields is None:
+            self.fields = [self.model._meta.get_field_by_name(field)[0] for field in self.field_names]
```

Comment:
this get_field_by_name should be adjusted with formalized model meta API I guess